python/graphstorm/config/argument.py

```diff
@@ -1393,8 +1501,6 @@ def exclude_training_targets(self):
     def gamma(self):
         """ Gamma for DistMult
         """
-        assert self.lp_decoder_type == BUILTIN_LP_DISTMULT_DECODER, \
```

Comment:
We do not use gamma with dot-product decoder

Reply:
we need to make a choice on how to verify the value when training/inference scripts start. either we don't verify it, or we only verify it when the lp decoder is distmult, or we don't enforce the check in the argument.

Reply:
My opinion is we can verify it when the lp decoder is distmult.

Reply:
that is very specific verification. we can do this for an argument, but my concern is that this will explode the difficulty of verifying arguments in the future.